dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/CommandMapper.xml

```diff
@@ -43,7 +43,7 @@
     <select id="queryCommandPageBySlot" resultType="org.apache.dolphinscheduler.dao.entity.Command">
         select *
         from t_ds_command
-        where id % #{masterCount} = #{thisMasterSlot}
+        where unix_timestamp(update_time) % #{masterCount} = #{thisMasterSlot}
```

Comment:
I am not sure if it is  a good idea to use update_time, if we support configure `db_step`, we can solve this problem by using `((id % db_step) % masterCount) = #{thisMasterSlot}`

Reply:
@ruanwenjun if user have master-slave database or AA database, the db_step may be different between each database instance

Reply:
AFAIW, the db_step should be same at master-slave database or AA database, otherwise, how to provide unique id?

Reply:
For example, if the database has two active-active instance, they will config mysql as below
set auto_increment_increment = 2
set first instance auto_increment_offset = 0
set second instance auto_increment_offset = 1

first database id will be 0, 2, 4, 6, 8
and second database id will be 1, 3, 4, 7, 9

Reply:
If there is new database setup, means there is a new ds cluster deploy? So you don't want to restart the old ds cluster?

I just worry `md5` is not supported by all database. `where conv(left(md5(id), 4), 16, 10) % #{masterCount} = #{thisMasterSlot}`.

Can we select a batch command from database e.g. select * from t_ds_command limit 100.

And filter the command at memory? We can provide some filter strategies, e.g hash_by_id(hash by id)、md5_by_id(use md5 to change the id and then do hash).

Reply:
Sure, maybe only MYSQL has those functions.

Reply:
Ok, so we all agree filter the command which should be consumed at memory rather than using database to do this kind of work, then we can support complex logic to `shuffle` the command.